pkg/kubelet/kubelet_pods.go

```diff
@@ -538,12 +538,26 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string, enableServiceLinks bool) (map[
 		mappedServices = append(mappedServices, serviceMap[key])
 	}
 
-	for _, e := range envvars.FromServices(mappedServices) {
+	for _, e := range envvars.FromServices(mappedServices, kl.isMasterService) {
 		m[e.Name] = e.Value
 	}
 	return m, nil
 }
 
+// isMasterService checks if svc is the master service.
+func (kl *Kubelet) isMasterService(svc *v1.Service) bool {
+	return svc.Namespace == kl.masterServiceNamespace && masterService == svc.Name
+}
+
+// filterService returns true for a service if
+// its ClusterIP is not "None" or empty
+// OR
+// it's the master service AND its ExternalName is not empty.
+func (kl *Kubelet) filterService(svc *v1.Service) bool {
```

Comment:
I'm wondering if we could find a more descriptive function name than `filterService`, as `filterService` feels fairly generic to me. Without reading the comment, its not obvious to me what `filterService` does.

Reply:
I'm open to any naming suggestions.

Reply:
To help me understand, what is the common characteristic of services for which the ClusterIp is not None/empty or master services in which ExternalName is not empty?

In other words, if you had to give a name to those types of services, what would it be?

Reply:
Renamed `filterService` to `valdiateService` as it's used for validation.